### PR TITLE
fix precision issue in Float64^Float64.

### LINKED ISSUE #44525: Difference (regression?) in floating point pow on 1.8
Looking at some PkgEval logs a failure that occurs comes up from a changed result for `2^111.94`. This used to be computed as

```jl
julia> v = 2^111.94
4.9807840732777406e33
```

which is in agreement with e.g. Python. The current value is

```jl
julia> 2^111.94
4.980784073277717e33
```

which to my naive eye seems to be significantly worse:

```jl
julia> v = 2^111.94; bv = big"2"^111.94; (v - bv) / bv # 1.7
2.09394118307...e-17

julia> v = 2^111.94; bv = big"2"^111.94; (v - bv) / bv # 1.8
-4.72427549745...e-15
```

### PR DESCRIPTION
**Pull request by oscardssmith**
Closes https://github.com/JuliaLang/julia/issues/44525
this only costs 1 `fma` more, so the performance impact is minimal.

### CODE REVIEW

**Comment by oscardssmith** [comment]
I've done local testing on 2^22 random inputs to confirm correctness. For uniform x,y (x in -0:2000, y in -100:100), mean error is now .257 ULP, maximum error found is 1.36 ULP, and only `1e-5` results have error greater than 1 ULP.

**Comment by KristofferC** [comment]
How about `10.0^-3` with this?

**Comment by oscardssmith** [comment]
Unchanged, but that is actually a bugfix. In 1.7, `10^-3 ` returned a different result than `10.0^-3` due to literal pow stuff.

**Comment by oscardssmith** [comment]
Also, what do you want in terms of tests for this? It's much harder to exhaustively test this than single argument functions because there are way more possibilities.

**Comment by ViralBShah** [comment]
Would be nice to have a slightly more descriptive issue title.

**Comment by giordano** [comment]
> 10^-3 returned a different result than 10.0^-3 due to literal pow stuff.

But now it returns a different results than `BigFloat(10.0) ^ -3`
```
julia> Float64(BigFloat(10.0) ^ -3)
0.001
```

**Comment by oscardssmith** [comment]
Yeah. The 2 options are to make `x^-3` 3x slower and always within .51 ULP, or leave as is. I'd be fine going either way.

**Comment by KristofferC** [comment]
> returned a different result than 10.0^-3 due to literal pow stuff.

It kind of feels "embarrassing" to not be able to compute something like 10.0^-3 accurately anymore (this goes for both the literal pow and the floating point version). Any idea what other programming languages do here? Are we the odd one out or is this common?

**Comment by oscardssmith** [comment]
I've created https://discourse.julialang.org/t/poll-speed-vs-performance-for-float64-float64/77619 to gauge community opinion here. I'm happy to go either way.

**Comment by oscardssmith** [comment]
I've now fixed x^-3 to be more accurate after some democracy, and more careful analysis (I hadn't realized that the fast version's error could get above 2 ULPs). @KristofferC or anyone else that wants to this is ready to review.

**Comment by KristofferC** [comment]
Still need to add tests for all the cases this changed I think?

**Comment by oscardssmith** [comment]
tests added.

File: test/math.jl [hunk 1]
```diff
@@ -1323,6 +1317,29 @@ end
...
+        for _ in 1:2^16
+            x=rand(T)*100; y=rand(T)*200-100
+            got, expected = x^y, widen(x)^y
+            if isfinite(eps(T(expected)))
+                @test abs(expected-got) <= 1.3*eps(T(expected))
+            end
+        end
+    end
+    # test (-x)^y for y larger than typemax(Int)
+    @test T(-1)^floatmax(T) === T(1)
```
Comment (KristofferC) [inline test/math.jl:1336]: 👀

File: test/math.jl [hunk 2]
```diff
@@ -1323,6 +1317,29 @@ end
     end
 end
 
+@testset "pow" begin
+    for T in (Float16, Float32, Float64)
+        for x in (0.0, -0.0, 1.0, 10.0, 2.0, Inf, NaN, -Inf, -NaN)
+            for y in (0.0, -0.0, 1.0, -3.0,-10.0 , Inf, NaN, -Inf, -NaN)
+                @test T(x)^T(y) === T(big(x)^y)
```
Comment (KristofferC) [inline test/math.jl:1324]: Doesn't work for `NaN`?

File: test/math.jl (hunk 1 above)
Comment (KristofferC) [reply test/math.jl:1336]: `T` not defined

**Comment by oscardssmith** [comment]
The tests also caught a minor precision bug for `Float32^-Integer` so that is now fixed.

**Comment by KristofferC** [comment]
Seems to still fail CI.

**Comment by giordano** [comment]
You may have to rebase on `master`

**Comment by KristofferC** [comment]
Seems like some other test started failing now.

**Comment by oscardssmith** [comment]
this is surprisingly fiddly to get 100% correct.

**Comment by oscardssmith** [comment]
This appears to finally be correct. Thanks for catching it before release @KristofferC.

**Pull request merged**